sparse/coo/core.py

```diff
@@ -1572,28 +1560,294 @@ def _keepdims(original, new, axis):
     return new.reshape(shape)
 
 
-def _mask(coords, idx, shape):
-    if isinstance(idx, numbers.Integral):
-        return coords == idx
-    elif isinstance(idx, slice):
-        step = idx.step if idx.step is not None else 1
-        if step > 0:
-            start = idx.start if idx.start is not None else 0
-            stop = idx.stop if idx.stop is not None else shape
-            return (coords >= start) & (coords < stop) & \
-                   (coords % step == start % step)
+def _mask(coords, indices, shape):
+    indices = _prune_indices(indices, shape)
+
+    ind_ar = np.empty((len(indices), 3), dtype=np.intp)
+
+    for i, idx in enumerate(indices):
+        if isinstance(idx, slice):
+            ind_ar[i] = [idx.start, idx.stop, idx.step]
         else:
-            start = idx.start if idx.start is not None else (shape - 1)
-            stop = idx.stop if idx.stop is not None else -1
-            return (coords <= start) & (coords > stop) & \
-                   (coords % step == start % step)
-    elif isinstance(idx, Iterable):
-        mask = np.zeros(len(coords), dtype=np.bool)
-        for item in idx:
-            mask |= _mask(coords, item, shape)
+            ind_ar[i] = [idx, idx + 1, 1]
```

Comment:
Just checking, but this will never get other numpy arrays of indices or boolean values?

Reply:
Right now advanced indexing isn't supported but it will be in the future. That is, technically, advanced indexing.

Reply:
Is the organization of code as you've made it here easy to extend to advanced cases?

Reply:
I'm fairly certain how it will be done, but I haven't mapped it out in code yet. However, it requires `list[list[int]]` on Numba's end.